### PR TITLE
Improves the way custom class names are updated on a tree node.

### PR DESCRIPTION
**Pull request by nathanbowser**
For https://github.com/SpiderStrategies/Scoreboard/issues/21974

### CODE REVIEW

File: test/tree-api-test.js [hunk 1]
```diff
@@ -832,9 +832,10 @@ test('edit changes class name on a node', function (t) {
 
     t.equal(el.querySelector('.tree ul li.node.foo.bar .label').innerHTML, 'Huge Scorecard', '1001 has classnames of foo and bar')
 
-    tree.edit({
-      id: 1001
-    })
+    let n = tree.get(1001)
+    // To remove a class name, you have to do an edit with `patch: false`
+    delete n.className
+    tree.edit(n, {patch: false})
```
Comment (nathanbowser) [inline test/tree-api-test.js]: @vtarchon Here's the new approach, will this work for you for https://github.com/SpiderStrategies/Scoreboard/issues/21974

Comment (vtarchon) [reply test/tree-api-test.js]: yeah that works for me, thanks!

**Comment by nathanbowser** [comment]
@mattsgarlata and @vtarchon are you +1 on this PR?

**Comment by vtarchon** [comment]
+1

**Comment by mattsgarlata** [comment]
+1

**Pull request merged**